Create URIports Guide (#1672)
zeit/docs: pages/guides/client-side-reports-with-uriports-zeit-now.mdx

import Guide from '~/components/layout/guide'
import { InlineCode } from '~/components/text/code'
import Snippet from '~/components/snippet'
import Caption from '~/components/text/caption'
import Note from '~/components/text/note'
import Link from '~/components/text/link'
import DeploySection from '~/components/guides/deploy-section'

export const meta = {

>   title: 'Reports for Your Projects with URIports and ZEIT Now',

Reviewer: ```suggestion
  title: 'Error Reports for Your Projects with URIports and ZEIT Now',
```

Reviewer: Added 'Error' otherwise what kind of reports are these?

  description:

>     'Enable client-side reporting for your website or web app ZEIT Now projects to act fast on issues, fix errors, and improve security, instead of relying on user reports.',

Reviewer: ```suggestion
    'Enable client-side reporting for your website or web app deployed with ZEIT Now to act fast on issues, fix errors, and improve security, instead of relying on user reports.',
```

  published: '2020-02-27T14:20:44.265Z',
  authors: ['timothy'],
  url: '/guides/client-side-reports-with-uriports-zeit-now',
  editUrl: 'pages/guides/client-side-reports-with-uriports-zeit-now.mdx',
  lastEdited: '2020-02-27T14:29:56.000Z'
}

Support for client-side generated reports are the standard for modern web browsers, and enabling this for your projects is a great way to identify and resolve issues, and improve site security, without relying on your users as testers.

URIports makes these reports free and easy to set up. You will only need to add configure a few HTTP headers to start receiving reports.

[ Image of reports on URIports ]

## What Reports?

Setting up reports with URIports, you can receive a variety of reports to inform you of various cases, including;

- Crash: A visitor was unable to continue because the (mobile) browser crashed.
- Deprecation: A browser API or feature has been used which is expected to stop working in a future update to the browser.
- Intervention: A user-agent has not honored a request made, or javascript function called, by the website or web-based app (e.g., for security, performance, or user annoyance reasons.)

> - Network-Error-Logging (NEL): Receive reports on 4xx/5xx HTTP errors, and DNS/TLS related issues. For more information, see <https://www.uriports.com/blog/network-error-logging/>.

Reviewer: ```suggestion
- Network-Error-Logging (NEL): Receive reports on 4xx/5xx HTTP errors, and DNS/TLS related issues. For more information, see the [URIports documentation](https://www.uriports.com/blog/network-error-logging/).
```

> - Content Security Policy (CSP): Reports to help detect and mitigate certain types of attacks, like XSS and data injection attacks and thereby improve the security of the website or web-based apps.

Reviewer: ```suggestion
- Content Security Policy (CSP): Reports to help detect and mitigate certain types of attacks, like XSS and data injection, and thereby improve the security of the website or web-based apps.
```

## Step 1: Setting up a project on ZEIT Now

ZEIT Now makes it effortless to deploy any static framework and Serverless Function so that you can focus on your project.

Firstly, signup for a free ZEIT account, then using your project's Git repository, import and deploy your project:

<DeploySection meta={meta} />

Alternatively, you can learn about setting up your project on ZEIT Now or use one of our many Quickstarts from [the ZEIT Now introduction documentation](/docs).

Once deployed, you will receive a default production URL for your project which you will need in the setup of your URIports account. Alternatively, you can [assign a Custom Domain](/docs/v2/custom-domains) to your project.

[ production domain example image? ]

## Step 2: Setting up URIports and HTTP Headers

URIports collects, aggregates, and enriches all of the reports listed above and presents them in a uniform, easy to navigate interface. With built-in (email and push) notifications, URIports keeps you informed on issues that require attention without the need to regularly log in.

> Create an endpoint for your reports by setting up a free URIports account here: <https://app.uriports.com/auth/create/>

Reviewer: ```suggestion
Create an endpoint for your reports by [setting up a free URIports account](https://app.uriports.com/auth/create/).
```

        },
        {
          "key": "Feature-Policy-Report-Only",
          "value": "geolocation 'none'; camera 'none'; fullscreen *; payment 'self'"
        }
      ]
    }
  ]
}
```

<Caption>
  HTTP headers to tell your visitor's browser to send reports to your endpoint.
</Caption>

<Note>
  You can learn more about URIports reporting types and how to use them from
  their{' '}
  <Link href="https://www.uriports.com/getting-started">
    Getting Started guide
  </Link>
  .
</Note>

## Step 3: Deploy and Receive Reports

With a ZEIT Now project created and URIports setup for your project, you can now deploy your changes and start receiving reports.

For projects that were deployed from a Git repository, all you need is to push to the default branch (usually `master`) and you will get a fresh production deployment.

Alternatively, run `now --prod` with Now CLI from your project's root.

That's all! You will now receive reports from your users and be able to act quickly on any issue that arises.

## Avoiding Flooding Your URIports Account

Some report types can potentially flood your URIports account with reports and burn a lot of your quota if they are configured incorrectly.

> Following is a few suggestions to help you stop flooding:

Reviewer: ```suggestion
The following is a few suggestions to help you stop flooding:
```
